quickwit/quickwit-query/src/query_ast/wildcard_query.rs

```diff
@@ -0,0 +1,186 @@
+                    "raw"
+                };
+                let mut text_analyzer = tokenizer_manager
+                    .get(analyzer_name)
+                    .with_context(|| "missing built-in tokenizer")?;
+                let mut token_stream = text_analyzer.token_stream(prefix);
+                let mut tokens = Vec::new();
+                token_stream.process(&mut |token| {
+                    let term: Term = Term::from_field_text(field, &token.text);
+                    tokens.push(term);
+                });
+                let term = tokens
+                    .pop()
+                    .with_context(|| "wildcard query generated no term")?;
+                if !tokens.is_empty() {
+                    return Err(InvalidQuery::Other(anyhow::Error::msg(
+                        "wildcard query generated more than one term",
+                    )));
+                }
+                Ok((field, term))
+            }
+            FieldType::JsonObject(json_options) => {
+                let text_field_indexing =
+                    json_options.get_text_indexing_options().ok_or_else(|| {
+                        InvalidQuery::SchemaError(format!(
+                            "field {} is not full-text searchable",
+                            field_entry.name()
+                        ))
+                    })?;
+                let tokenizer_name = text_field_indexing.tokenizer();
+                let analyzer_name = if tokenizer_manager
+                    .get_does_lowercasing(tokenizer_name)
+                    .with_context(|| {
+                        format!("no tokenizer named `{}` is registered", tokenizer_name)
+                    })? {
+                    "lowercase"
+                } else {
+                    "raw"
+                };
+                let mut text_analyzer = tokenizer_manager
+                    .get(analyzer_name)
+                    .with_context(|| "missing built-in tokenizer")?;
+                let mut token_stream = text_analyzer.token_stream(prefix);
+                let mut tokens = Vec::new();
+                let mut term = Term::with_capacity(100);
+                let mut json_term_writer = JsonTermWriter::from_field_and_json_path(
+                    field,
+                    json_path,
+                    json_options.is_expand_dots_enabled(),
+                    &mut term,
+                );
+
+                token_stream.process(&mut |token| {
+                    json_term_writer.set_str(&token.text);
+                    tokens.push(json_term_writer.term().clone());
+                });
+                let term = tokens
+                    .pop()
+                    .with_context(|| "wildcard query generated no term")?;
+                if !tokens.is_empty() {
+                    return Err(InvalidQuery::Other(anyhow::Error::msg(
+                        "wildcard query generated more than one term",
+                    )));
+                }
```

Comment:
could be nice to isolate a check function. 

Also the errors are a bit misleading, `tokenzing wildcard query generated no term`? same for the second error message